src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -592,9 +592,19 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
 
                 int duplicateEmailIndex = getDuplicateEmailIndex(student.email, studentList);
                 if (duplicateEmailIndex != -1) {
-                    invalidityInfo.add(duplicateEmailInfo(sanitizedLine, linesArray[duplicateEmailIndex + 1]));
+                    // if student is valid then this is the only exception message added,
+                    // thus require the sanitizedLine to be prepended
+                    if (invalidityInfo.isEmpty() || student.isValid()) {
+                        invalidityInfo.add(sanitizedLine.concat("##").concat(
```

Comment:
I'm not 100% sure what this `"##"` is doing but skimming through the code, my guess is that you are using it to delimit different error messages. This is definitely not the right way to do. If you need a list of sth, use an `ArrayList`.